.circleci/bin/test-airflow-image.py

```diff
@@ -106,6 +106,79 @@ def test_astronomer_airflow_check_version(webserver):
         "astronomer-airflow-version-check module must be greater than 1.0.0"
 
 
+def test_airflow_connections(scheduler):
+    """Test Connections can be added and deleted"""
+    test_conn_uri = "postgresql://postgres_user:postgres_test@1.1.1.1:5432"
+    test_conn_id = "test"
+
+    # Assert Connection can be added
+    assert f"Successfully added `conn_id`={test_conn_id} : {test_conn_uri}" in scheduler.check_output(
+        f'airflow connections -a --conn_uri "{test_conn_uri}" --conn_id {test_conn_id}')
+
+    # Assert Connection can be removed
+    assert f"Successfully deleted `conn_id`={test_conn_id}" in scheduler.check_output(
+        f'airflow connections -d --conn_id {test_conn_id}')
+
+
+def test_airflow_variables(scheduler):
+    """Test Variables can be added, retrieved and deleted"""
+    # Assert Variables can be added
+    assert "" in scheduler.check_output("airflow variables --set test_key test_value")
+
+    # Assert Variables can be retrieved
+    assert "test_value" in scheduler.check_output("airflow variables --get test_key")
+
+    # Assert Variables can be deleted
+    assert "" in scheduler.check_output("airflow variables --delete test_key")
+
+
+def test_list_dags(scheduler):
+    """
+    Create Example DAG and add it to Scheduler POD
+    """
+    airflow_list_dags_output = scheduler.check_output("airflow list_dags -r")
+    assert "Number of DAGs: 1" in airflow_list_dags_output
+    assert "example_dag" in airflow_list_dags_output
+
+
+def test_airflow_trigger_dags(scheduler):
+    """Test Triggering of DAGs & Pausing & Unpausing Dags"""
+
+    timeout = 180
+    sleep_count = 0
+    sleep_time_between_polls = 5
+    while scheduler.run("airflow pause example_dag").exit_status != 0:
```

Comment:
Can pause ever fail?!

Reply:
Yes, I was bitten by this. This happens when the `dag` does not exist in the DagBag.

Reply:
But how would retrying help that? Cos `airflow pause` loads the dagbag from scratch doesn't it? Or does the CLI use s10n too?

Reply:
True, This was needed why I was adding the file by exec'ing but not needed anymore